File: code/mathematical-algorithms/catalan_number/Catalan_Number.cpp
//First n Catalan numbers using Dynamic Programming!!!

#include <iostream>
using namespace std;
// Part of Cosmos by OpenGenus Foundation

#define ll long long int 

int main()
{
	int n;
	cin>>n;
	ll catalan_array[10000];
	int c0=1;
	cout<<c0<<endl;
	catalan_array[0]=c0;
	cout<<catalan_array[0]<<endl;
	for(int i=1;i<n-1;i++)
	{
	    catalan_array[i]=((2*(2*(i)+1))*(catalan_array[i-1]))/(i+2);
	    cout<<catalan_array[i]<<endl;
	}
	
}

File: code/mathematical_algorithms/src/catalan_number/catalan_number.py
# Part of Cosmos by OpenGenus Foundation


def catalan(n):

    if n <= 1:
        return 1
    res = 0
    for i in range(n):
        res += catalan(i) * catalan(n - i - 1)
    return res


for i in range(10):
    print(catalan(i), emd=" ")
